src/main/java/com/bench/stockmanagement/SellingHandler.java

```diff
@@ -2,51 +2,74 @@
+    private final SellingRepository sellingRepository;
 
     @Autowired
-    public SellingHandler(SellingReader reader, SellingMapper mapper, SellingService sellingService) {
+    public SellingHandler(SellingReader reader, SellingMapper mapper, SellingRepository sellingRepository) {
         this.reader = reader;
         this.mapper = mapper;
-        this.sellingService = sellingService;
+        this.sellingRepository = sellingRepository;
     }
 
-    public void loadSoldItems() {
+    // Save sold items
+    public Mono<Result> loadSoldItems() {
         List<Receipt> soldItems = reader.readSoldItems();
         List<DBReceipt> dbReceipts = mapper.mapToDB(soldItems);
-        sellingService.saveSoldItems(dbReceipts);
+        return Flux.fromIterable(dbReceipts)
+                .doOnNext(sellingRepository::saveSoldItems)
+                .then(Mono.just(SUCCESS))
+                .onErrorReturn(FAIL);
     }
 
-    public Receipt getReceipt(String receiptNumber) {
-        List<DBReceipt> receipts = sellingService.getAReceipt(receiptNumber);
-        return mapper.mapReceipt(receipts);
+    // Get a receipt by receipt number
+    public Mono<Receipt> getReceipt(String receiptNumber) {
+        SdkPublisher<Page<DBReceipt>> receiptById = sellingRepository.getReceiptById(receiptNumber);
+        return Flux.from(receiptById)
+                .log()
+                .map(Page::items)
+                .map(mapper::mapReceipt)
+                .singleOrEmpty();
     }
 
-    public List<Receipt> getAllReceipt() {
-        List<DBReceipt> allReceipt = sellingService.getAllReceipt();
-        return mapper.mapReceipts(allReceipt);
+    // Get all receipt
+    public Flux<Receipt> getAllReceipt() {
+        return Flux.from(sellingRepository.getAllReceipt()).log().map(Page::items).flatMapIterable(mapper::mapReceipts);
     }
 
-    public List<Receipt> getSoldItemBetween(String startDate, String endDate) {
-        List<DBReceipt> soldItems = sellingService.getSoldItemsByDate(startDate, endDate);
-        return mapper.mapReceipts(soldItems);
+    // Get items from all receipts by item number
+    public Flux<SoldItem> getItems(String itemNumber) {
+        return Flux.from(sellingRepository.getSoldItemsByItemNumber(itemNumber)).log()
+                .map(Page::items)
+                .flatMapIterable(mapper::getItems);
     }
 
-    public List<SoldItem> getItems(String itemNumber) {
-        List<DBReceipt> soldItemsByItemNumber = sellingService.getSoldItemsByItemNumber(itemNumber);
-        return mapper.getItems(soldItemsByItemNumber);
+    // Get receipts between dates which contain the given item numbers
+    public Flux<Receipt> getSoldItemBetween(String startDate, String endDate, String... itemNumber) {
+        return Flux.fromIterable(List.of(itemNumber)).log()
+                .map(in -> sellingRepository.getSoldItemByDate(in, startDate, endDate))
+                .flatMap(Flux::from)
```

Comment:
This is just nitpicking but I think it makes sense to merge these two steps:
```suggestion
                .flatMap(in -> Flux.from(sellingRepository.getSoldItemByDate(in, startDate, endDate)))
```